docs/shared/concepts/sync-generators.md

```diff
@@ -0,0 +1,46 @@
+# Sync Generators
+
+In Nx 19.7, you can use sync generators to update code before a task is run or before code is sent to CI. Sync generators are designed to perform maintenance work that is required for a particular task or for CI.
+
+Sync Generator Examples:
+
+- Ensure code is formatted in a specific way before CI is run
+- Update a custom CI script with binning strategies based on the current project graph
+- Update TypeScript config files with project references based on the current project graph
+
+## Task Sync Generators
+
+Sync generators can be associated with a particular task. Nx will ensure that the sync generator is run before the task is executed. This is similar to the `dependsOn` property, but for generators instead of task dependencies.
+
+To [register a generator](/extending-nx/recipes/register-sync-generator) as a sync generator for a particular task, add the generator to the `syncGenerators` property of the task configuration.
+
+## Global Sync Generators
+
+Global sync generators are not associated with a particular task and are executed only when the `nx sync` or `nx sync:check` command is explicitly run. They are [registered](/extending-nx/recipes/register-sync-generator) in the `nx.json` file with the `sync.globalGenerators` property.
+
+## Use the Project Graph in a Sync Generator
```

Comment:
This section doesn't show you how to use the project graph in a sync generator?

Reply:
Correct.  I updated the title to better describe what this section is about.  How to use the project graph should go in the Create a Sync Generator recipe, not in the concept page.